sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -172,7 +172,8 @@ abstract class Optimizer(sessionCatalog: SessionCatalog)
     Batch("Object Expressions Optimization", fixedPoint,
       EliminateMapObjects,
       CombineTypedFilters,
-      ObjectSerializerPruning) :+
+      ObjectSerializerPruning,
+      ReassignLambdaVariableID) :+
```

Comment:
This rule should be applied only `Once` instead of `fixedPoint`? otherwise, per-query unique ID might conflict?

Reply:
The newly generated unique IDs are negative, so this rule is idempotent because it only catches positive IDs.

Reply:
But if the rule applied twice with including the positive IDs in the second loop for some reason, the new ID starts from -1 again, then it conflicts the first -1, I think.

Reply:
The IDs should all be positive or negative. Let me add some checks to ensure that.

Reply:
Thanks, sounds great with the check.